### DISCUSSION TITLE
Tokio v1.25.0

### DISCUSSION THREAD

**Discussion by Darksonn** [tokio-rs/tokio#5409] [Announcements]
# 1.25.0 (January 28, 2023)

### Fixed

- rt: fix runtime metrics reporting ([#5330])

### Added

- sync: add `broadcast::Sender::len` ([#5343])

### Changed

- fs: increase maximum read buffer size to 2MiB ([#5397])

[#5330]: https://github.com/tokio-rs/tokio/pull/5330
[#5343]: https://github.com/tokio-rs/tokio/pull/5343
[#5397]: https://github.com/tokio-rs/tokio/pull/5397

<hr /><em>This discussion was created from the release <a href='https://github.com/tokio-rs/tokio/releases/tag/tokio-1.25.0'>Tokio v1.25.0</a>.</em>

**Comment by ClementNerma** [comment]
This may be a stupid question but why is this a minor release? Shouldn't it be a patch release since it contains so little changes?

**Reply by Altair-Bueno** [reply]
Probably due to #5397. Changing the buffer size could have unexpected consequences, specially if your system has been manually fine tuned

**Reply by Noah-Kennedy** [reply]
A patch release is only for bugfixes, not new features.

**Reply by ClementNerma** [reply]
I've seen other projects bump the patch version for very minor improvements so I thought it was the norm. My mistake!

**Reply by Darksonn** [reply]
The `broadcast::Sender::len` addition alone would also be enough to make it a minor release. It's a new feature.

Doing it this way allows us to make additional patch releases of 1.24.x in the future if that version contains any bugs, without forcing people to pick up the changes in this release together with the bugfix. (1.24.x isn't a LTS release, so we're unlikely to make futher patch releases of it, but the point stands.)
